src/__tests__/__helpers__/mockClerkServer.ts

```diff
@@ -0,0 +1,141 @@
+import * as clerk from "@clerk/nextjs/server";
+export type User = NonNullable<Awaited<ReturnType<typeof clerk.currentUser>>>;
+export type AuthObject = ReturnType<typeof clerk.getAuth>;
+
+/**
+ * Sets the return value of `currentUser()` on the server
+ */
+export function setCurrentUser(user: User | null) {
+  jest.spyOn(clerk, "currentUser").mockResolvedValue(user);
+}
+
+/**
+ * Sets the return value of `getAuth()` on the server
+ */
+export function setGetAuth(auth: AuthObject) {
+  jest.spyOn(clerk, "getAuth").mockReturnValue(auth);
+}
+
+/**
+ * Installs a partial mock of `clerkClient()` for use on the server
+ * to make requests to Clerk's API
+ */
+export function installMockClerkClient() {
+  const mockClerkClient = {
+    users: {
+      updateUserMetadata: () => Promise.resolve(mockServerUser),
+      getUser: () => Promise.resolve(mockServerUser),
+    },
+  } as unknown as ReturnType<typeof clerk.clerkClient>;
+  jest.spyOn(clerk, "clerkClient").mockReturnValue(mockClerkClient);
+  return mockClerkClient;
+}
```

Comment:
This is probably going to be quite common, so a utility to install a mock implementation will be handy